File: hackerrank/easy/0000-grading-students/swift/solution.swift
// solution.swift

func gradingStudents(grades: [Int]) -> [Int] {

    func gradeRoundUp(grade: Int) -> Int {
        let nextMultipleOfFive = grade + (5 - grade % 5)
        return (nextMultipleOfFive - grade < 3) ? nextMultipleOfFive : grade
    }

    func gradeCheck(grade: Int) -> Int {
        return (grade < 38) ? grade : gradeRoundUp(grade: grade)
    }

    return grades.map { gradeCheck(grade: $0) }
}

// --- demo run ---
let input = [73, 67, 38, 33]
let result = gradingStudents(grades: input)
print(result)   // [75, 67, 40, 33]

File: hackerrank/easy/0000-grading-students/js/solution.mjs
// solution.mjs
export function gradingStudents(grades) {
  return grades.map(g => {
    if (g < 38) return g;
    const next = Math.floor(g / 5) * 5 + 5;
    const diff = next - g;
    return diff < 3 ? next : g;
  });
}

// --- demo run ---
if (import.meta.url === `file://${process.argv[1]}`) {
  const input = [73, 67, 38, 33];
  const result = gradingStudents(input);
  console.log(result); // [75, 67, 40, 33]
}
